anchore_engine/analyzers/hints.py

```diff
@@ -0,0 +1,407 @@
+                        self.name
+                    )
+                )
+
+    def normalize(self):
+        self.validate_files()
+
+    def to_dict(self) -> dict:
+        result = super().to_dict()
+        result.update(
+            {
+                "license": self.license,
+                "files": self.files,
+                "metadata": self.metadata,
+                "location": self.location,
+            }
+        )
+        return result
+
+
+class DebianHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, "dpkg")
+        self.license = ensure_str(pkg.get("license", ""))
+        self.arch = ensure_str(pkg.get("arch", "x86_64"))
+        self.release = ensure_str(pkg.get("release", ""))
+        self.source = ensure_str(pkg.get("source", ""))
+        self.size = ensure_str(str(pkg.get("size", "0")))
+
+    def normalize(self):
+        if not self.source:
+            self.source = "%s-%s" % (self.name, self.version)
+        self.release = "N/A"
+
+    def to_dict(self) -> dict:
+        result = super().to_dict()
+        result.update(
+            {
+                "license": self.license,
+                "arch": self.arch,
+                "release": self.release,
+                "sourcepkg": self.source,
+                "size": self.size,
+            }
+        )
+        return result
+
+
+class AlpineHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, "APKG")
+        self.license = ensure_str(pkg.get("license", ""))
+        self.arch = ensure_str(pkg.get("arch", "x86_64"))
+        self.release = ensure_str(pkg.get("release", ""))
+        self.source = ensure_str(pkg.get("source", ""))
+        self.size = ensure_str(str(pkg.get("size", "0")))
+        self.files = pkg.get("files", [])
+
+    def normalize(self):
+        if not self.release:
+            try:
+                (v, r) = self.version.split("-", 2)
+            except ValueError:
+                raise HintsError(
+                    "hints package version for hints package ({}) is not valid for APKG package type".format(
+                        self.name
+                    )
+                )
+
+            self.release = r
+            self.version = v
+
+        if not self.source:
+            self.source = self.name
+
+    def to_dict(self) -> dict:
+        result = super().to_dict()
+        result.update(
+            {
+                "license": self.license,
+                "arch": self.arch,
+                "release": self.release,
+                "sourcepkg": self.source,
+                "size": self.size,
+                "files": self.files,
+            }
+        )
+        return result
+
+
+class GemHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, ensure_str(pkg.get("type", "gem")).lower())
+        self.version = self.get_list_value(pkg, "version")
+        if len(self.version) > 0:
```

Comment:
`self.version` is a list, so you can evaluate it as a _truthy_ value:

```suggestion
        if self.version:
```

Reply:
fixed here and in npmhint too